src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseCompositeFormat.cs

```diff
@@ -0,0 +1,230 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Performance
+{
+    using static MicrosoftNetCoreAnalyzersResources;
+
+    /// <summary>
+    /// CA1862: <inheritdoc cref="UseCompositeFormatTitle"/>
+    /// </summary>
+    /// <remarks>
+    /// Roslyn already provides a refactoring for finding string.Format calls with literal string formats
+    /// and converting them to use string interpolation. This analyzer instead focuses on non-literal / const
+    /// arguments.
+    /// </remarks>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class UseCompositeFormatAnalyzer : DiagnosticAnalyzer
+    {
+        internal static readonly DiagnosticDescriptor UseCompositeFormatRule = DiagnosticDescriptorHelper.Create("CA1862",
+            CreateLocalizableResourceString(nameof(UseCompositeFormatTitle)),
+            CreateLocalizableResourceString(nameof(UseCompositeFormatMessage)),
+            DiagnosticCategory.Performance,
+            RuleLevel.IdeSuggestion,
```

Comment:
Isn't the API review approved it with Hidden level?
```suggestion
            RuleLevel.IdeHidden_BulkConfigurable,
```